DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Reports/ReportsController.cs

```diff
@@ -97,19 +142,29 @@ public IEnumerable<ActivityDataRowModel> GetGraphData()
             ReportInterval reportInterval
         )
         {
-            var parsedStartDate = DateTime.Parse(startDate);
-            var parsedEndDate = DateTime.Parse(endDate);
-
             var centreId = User.GetCentreId();
             var adminId = User.GetAdminId()!.Value;
             var adminUser = userDataService.GetAdminUserById(adminId)!;
 
+            var parsedStartDate = DateTime.Parse(startDate);
+            if (parsedStartDate < activityService.GetStartOfActivityForCentre(centreId))
+            {
+                return new NotFoundResult();
+            }
+
+            if (DateTime.TryParse(endDate, out var parsedEndDate) &&
+                (parsedEndDate < parsedStartDate || parsedEndDate > DateTime.Now))
+            {
+                return new NotFoundResult();
+            }

```

Comment:
Could we put this into a service? Potentially something like `getValidatedUsageStatsDateRange(startDateString, endDateString)` that returns `(startDate, endDate?)?`
Then we can return NotFound if the tuple is null. And the date validation will also mean that we can just pass in endDate to the ActivityFilterData and keep all the relevant logic in service (with nice small unit tests).